plugins/metadata/services/MetadataService.php

```diff
@@ -78,6 +78,16 @@ function addAction($metadataProfileId, $objectType, $objectId, $xmlData)
 		}
 		
 		$objectType = kPluginableEnumsManager::apiToCore('MetadataObjectType', $objectType);
+
+		$limitEntry = $this->getKs()->getLimitEntry();
+		if ($limitEntry) {
+			$peer = kMetadataManager::getObjectPeer($objectType);
+			$entry = $peer->getEntry($objectId);
+			if ($limitEntry && $entry && $entry->getId() != $limitEntry) {
```

Comment:
In case we failed to retrieve the entry (entitlement, etc ...) the if will not be true and we will no enforce the limitEntry.
I think than in cases we faile to retrieve the entry we should throw exception as well.